src/adapters/refract/transformSections.coffee

```diff
@@ -2,37 +2,65 @@ _ = require('./helper')
 
 blueprintApi = require('../../blueprint-api')
 getDescription = require('./getDescription')
+
 transformResources = require('./transformResources')
+transformResource = require('./transformResource')
+
 
-module.exports = (element) ->
+module.exports = (parentElement) ->
   resourceGroups = []
-  resourceGroupsElements = _.chain(element)
-                    .content()
-                    .filter({element: 'category', meta: {classes: ['resourceGroup']}})
-                    .value()
-
-  # If we can't find any resource group we can assume that there are Resources directly
-  if resourceGroupsElements.length is 0
-    resourceGroups.push(
-      new blueprintApi.Section({
-        name: ''
+
+  # List of Application AST Resources (= already transformed
+  # Refract Resources into the Application AST Resource).
+  applicationAstResources = []
+
+  _.forEach(parentElement.content, (element, index) ->
+    # If the element is a resource, transform it
+    # into the Application AST Resource and push it
+    # to a temporary array, we'll create an artificial
+    # resource group (section) for these resources
+    # later in the process.
+    if element.element is 'resource'
+      applicationAstResources = applicationAstResources.concat(
+        transformResource(element)
+      )
+
+    # Element is category (resource group most probably).
+    if element.element is 'category'
+      # First let's create an artificial resource group (section)
+      # for previous resources.
+      if applicationAstResources.length
+        resourceGroups.push(new blueprintApi.Section({
+          name: ''
+          resources: applicationAstResources
+        }))
+
+        applicationAstResources = []
```

Comment:
@smizell 

### Input

```
[
  Resource A
  Resource B
  Resource Group AA
  Resource C
]
```

### Loop

Run `#0`

  * It's a resource
  * Push the _Resource A_ to `applicationAstResources`

Run `#1`
  * It's a resource
  * Push the _Resource B_ to `applicationAstResources`

Run `#2`
  * It's a resource group
  * Take all resources (_Resource A_ and _Resource B_) in `applicationAstResources` and create a new resource group for them
  * Push that resource group to `resourceGroups`
  * Empty `applicationAstResources` (`= []`)
  * Push the _Resource Group AA_ to `resourceGroups`

Run `#3`
  * It's a resource
  * Push the _Resource C_ to `applicationAstResources`

End
  * Whoa, `applicationAstResources` contains some resources (_Resource C_)
  * Take resources (_Resource C_) in `applicationAstResources` and create a new resource group
  * Push that resource group to `resourceGroups`
  * Return `resourceGroups`